README.md

```diff
@@ -1,6 +1,57 @@
-# LLVM-based Solidity Compiler Toolchain
+<div align="center">
+  <img src=".github/assets/logo.png" alt="solx logo" />
+</div>
 
-This repository contains an LLVM-based Compiler Toolchain for Solidity and Yul.
+### `solx` is an experimental compiler for Solidity
+
+The goal of the project is to make state-of-the-art optimizing compilers a mainstream for Web3.
+
+We want to:
+
+- Make Ethereum more efficient.
+- Eliminate the need to write inline assembly.
+- Provide industry-grade tooling, such as interactive debugger.
+
+## Installation
+
+You can get existing pre-releases from the [Releases](https://github.com/matter-labs/solx/releases) GitHub page.
+Or, you can take a build used in [solx_demo](https://github.com/popzxc/solx_demo):
+
+- [Linux/AMD64](https://github.com/matter-labs/solx/releases/download/eb46690/solx-linux-amd64-gnu-test-build-06)
+- [Linux/Arm64](https://github.com/matter-labs/solx/releases/download/eb46690/solx-linux-arm64-gnu-test-build-06)
+- [MacOS](https://github.com/matter-labs/solx/releases/download/eb46690/solx-macosx-test-build-06)
+
+## Usage
+
+`solx` currently works by disguising itself as `solc` 0.8.29.
+
+For now, the recommended way to use `solx` is via [foundry](https://github.com/foundry-rs/foundry).
+You can download the binary, and provide path to it as `solc_version`, for example:
+
+```toml
+[profile.solx]
+solc_version = "/path/to/solx"
+```
+
+It might work with `hardhat` as well, but this has not been tested yet.
+
+Otherwise, the interface is _mostly_ compatible with `solc`, so you can use it via CLI or standard JSON.
+
+## Demo
+
+Check [this repository](https://github.com/popzxc/solx_demo) to see a demo of the current state of the compiler.
+
+## Architecture
+
+`solx` consists of three main parts:
+
+- `solx` executable, present in this repository.
+- [`era-solidity`](https://github.com/matter-labs/era-solidity/), a fork of the [solidity compiler](https://github.com/ethereum/solidity),
+  which contains the compiler front-end. Despite the repository name, it is not directly related to either ZKsync or ZKsync Era.

```

Comment:
solx also contains a part of FE (Yul and EVMLA lowering), but we can probably omit it for the sake of simplicity.